backend/ohq/serializers.py

```diff
@@ -447,3 +449,105 @@ def create(self, validated_data):
     def update(self, instance, validated_data):
         validated_data["author"] = self.context["request"].user
         return super().update(instance, validated_data)
+
+
+class RuleSerializer(serializers.ModelSerializer):
+    """
+    Serializer for rules
+    """
+
+    class Meta:
+        model = Rule
+        fields = ("frequency",)
+
+
+class EventSerializer(serializers.ModelSerializer):
+    """
+    Serializer for events
+
+    All times are converted to UTC+0
+    """
+
+    rule = RuleSerializer(required=False)
+    course_id = serializers.IntegerField(required=False)
+
```

Comment:
Just read this stack overflow post: https://stackoverflow.com/questions/53735960/django-rest-framework-how-to-make-field-required-read-only-only-for-update-ac/53736259

Seems like if we use this, we can make certain fields required for POST like the course id